src/jrd/InitCDSLib.h

```diff
@@ -0,0 +1,70 @@
+/*
+ *	PROGRAM:	JRD Access Method
+ *	MODULE:		InitCDSLib.h
+ *	DESCRIPTION:	support for correct usage of CDS library by the engine
+ *
+ *  The contents of this file are subject to the Initial
+ *  Developer's Public License Version 1.0 (the "License");
+ *  you may not use this file except in compliance with the
+ *  License. You may obtain a copy of the License at
+ *  http://www.ibphoenix.com/main.nfs?a=ibphoenix&page=ibp_idpl.
+ *
+ *  Software distributed under the License is distributed AS IS,
+ *  WITHOUT WARRANTY OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing rights
+ *  and limitations under the License.
+ *
+ *  The Original Code was created by Vladyslav Khorsun for the
+ *  Firebird Open Source RDBMS project.
+ *
+ *  Copyright (c) 2022 Vladyslav Khorsun <hvlad@users.sourceforge.net>
+ *  and all contributors signed below.
+ *
+ *  All Rights Reserved.
+ *  Contributor(s): ______________________________________.
+ *
+ */
+
+#ifndef FB_INIT_CDSLIB_H
+#define FB_INIT_CDSLIB_H
+
+#include "../common/classes/alloc.h"
+
+// Uncomment to write final memory usage stats into firebird.log.
+// See ~InitCDS() and ~InitPool() in CCH
+
+//#define DEBUG_CDS_MEMORY
+
+namespace Jrd
+{
+
+class InitCDS
+{
+public:
+
+	explicit InitCDS(MemoryPool&);
+	~InitCDS();
+
+	// Creates memory pool that will not be deleted until cds finish its work.
```

Comment:
How is this relevant for current master, which have the external pool?
External pool and special pool destroy logic was added just for these things and it works very weel with boost.test.
Did you tested the code against recent codebase without this workaround?

Reply:
I'll look at it. 
The code was written on the v4 code base initially. And I going to maintain it in my custom tree at least for some time.
What problem do you see with this code ?

Reply:
> What problem do you see with this code ?

As said, a not more needed workaround since the solution for the problem was already implemented.

Reply:
The external pool give no benefits for CDS, AFAIU.

CDS allows full control on memory usage and code uses it.
Current code uses two pools:
- one allocated in InitCDS::InitCDS() and used for internal needs of CDS, and
- one allocated in InitPool::InitPool() (at cch.cpp) and used for list nodes (in hash table) only 
Note, there could be allocated very big number of list nodes and distinct pool allows to exactly control its memory usage and avoid fragmentation of pool memory.
~InitCDS() guarantees that pools destroyed after CDS terminated its operations.
Thus there is no need to allocate pools for CDS from "never freed" external pool.
While it is possible, it is not necessary.
And it will not avoid usage of InitCDS().
And I don't want to make all CDS allocations of single common (external) pool, if you offers exactly this.